kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionImpl.java

```diff
@@ -662,18 +662,16 @@ public List<DomainMetadata> getComputedDomainMetadatasToCommit() {
       }
 
       generateClusteringDomainMetadataIfNeeded();
-
-      if (domainsToAdd.isEmpty() && domainsToRemove.isEmpty()) {
-        // If no domain metadatas are added or removed, return an empty list. This is to avoid
-        // unnecessary loading of the domain metadatas from the snapshot (which is an expensive
-        // operation).
-        computedMetadatas = Optional.of(Collections.emptyList());
-        return Collections.emptyList();
-      }
-
       // Add all domains added in the transaction
       List<DomainMetadata> result = new ArrayList<>(domainsToAdd.values());
 
+      if (domainsToRemove.isEmpty()) {
```

Comment:
cc @huan233usc could you please add a test for this as part of your metrics test refactoring and new suite for domain metadata? Thank you!